plugin.go

```diff
@@ -11,12 +11,28 @@ type Logger interface {
 }
 
 type Plugin struct {
-	log *zap.Logger
+	log    *zap.Logger
+	config *Config
 }
 
-func (p *Plugin) Init(log Logger) error {
+type Configurer interface {
+	// UnmarshalKey takes a single key and unmarshal it into a Struct.
+	UnmarshalKey(name string, out any) error
+	// Has checks if a config section exists.
+	Has(name string) bool
+	// RRVersion is the roadrunner current version
+	RRVersion() string
+}
+
+func (p *Plugin) Init(cfg Configurer, log Logger) error {
 	p.log = log.NamedLogger(pluginName)
 
+	p.config = &Config{}
+	err := cfg.UnmarshalKey(pluginName, &p.config)
+	if err != nil {
+		return err
```

Comment:
It should be `errors.E(op, err)` where `op` is an operation, and `errors.E` is a wrapper. You may take a look at the other plugins.

Reply:
fixed